src/librustdoc/clean/types.rs

```diff
@@ -1966,11 +1967,67 @@ crate struct Static {
 }
 
 #[derive(Clone, PartialEq, Eq, Hash, Debug)]
-crate struct Constant {
-    crate type_: Type,
-    crate expr: String,
-    crate value: Option<String>,
-    crate is_literal: bool,
+crate enum Constant {
+    /// This is the wrapper around `ty::Const` for a non-local constant. Because it doesn't have a
+    /// `BodyId`, we need to handle it on its own.
+    TyConst { type_: Type, expr: String },
+    /// A constant (expression) that’s not an item or associated item. These are usually found
+    /// nested inside types (e.g., array lengths) or expressions (e.g., repeat counts), and also
+    /// used to define explicit discriminant values for enum variants.
+    Anonymous { type_: Type, body: BodyId },
+    /// Inlined constant.
+    Extern { type_: Type, def_id: DefId },
```

Comment:
This is specifically for non-local constants I thought? Or does `clean::inline` discard the BodyId? That doesn't seem right.
```suggestion
    /// A constant from a different crate.
    Extern { type_: Type, def_id: DefId },
```